src/amp-story-player/amp-story-player-impl.js

```diff
@@ -486,6 +487,36 @@ export class AmpStoryPlayer {
     }
   }
 
+  /**
+   * Navigates stories given a number.
+   * @param {number} storyDelta
+   * @private
+   */
+  go(storyDelta) {
+    if (this.currentIdx_ + storyDelta > this.stories_.length) {
+      throw new Error('Out of Story range.')
+    }
+    if (this.currentIdx_ + storyDelta < 0) {
+      throw new Error('Out of Story range.')
+    }
+    
+    if (storyDelta === 0) {
+      return;
+    }
+    if (storyDelta > 0) {
+      for (let i = 0; i < storyDelta; i++) {
+        this.next_();
```

Comment:
@Enriqe will this cause multiple allocations/deallocations of iframes?

Reply:
Yeah, especially if the given delta is a story further in the array that has many stories in between without an iframe. I think we should be able to optimize this by using the `show()` method instead. @bettyashley